checklist/src/checklist_bot.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Octokit} from '@octokit/rest';
+
+import {GitHub} from './github';
+import {ILogger} from './types';
+import {checklistSectionBody, commentBody} from './template';
+
+const path = 'extensions';
+
+/** A GitHub bot which posts checklists on pull requests. */
+export class ChecklistBot {
+  readonly github: GitHub;
+
+  /** */
+  constructor(
+    client: Octokit,
+    private org: string,
+    private logger: ILogger = console
+  ) {
+    this.github = new GitHub(client, org, logger);
+    this.logger.info(`ChecklistBot initialized for ${this.org}`);
```

Comment:
Nit: This will probably be a repo-bot rather than an org-wide bot. Can address in a later stage of review.